src/engine/Sprite.cxx

```diff
@@ -33,8 +34,9 @@ void Sprite::render() const
         SDL_SetTextureColorMod(m_SpriteData[currentLayer].texture, highlightColor.r, highlightColor.g, highlightColor.b);
       }
 
-      if (transparentSprite)
+      if (currentLayer == Layer::ZONE)
       {
+        unsigned char alpha = (1 - Settings::instance().zoneLayerTransperancy) * 255;
```

Comment:
avoid calculations or anything that can introduce load in sprite.cxx. the slightest load will have a huge impact on FPS. 
move the alpha value calculation to a different place and calculate it once when it's changed. then use the stored variable.

Reply:
done, moved to MapNode